src/Tribe/Tickets.php

```diff
@@ -239,14 +239,55 @@ public function get_event_for_ticket( $possible_ticket ) {
 		abstract protected function get_tickets( $event_id );
 
 		/**
+		 * Get attendees by id and associated post type
+		 * or default to using $post_id
+		 *
+		 * @param      $post_id
+		 * @param null $post_type
+		 *
+		 * @return array|mixed
+		 */
+		abstract public function get_attendees_by_id( $post_id );
+
+		/**
 		 * Get all the attendees (sold tickets) for an event
+		 *
 		 * @abstract
 		 *
-		 * @param $event_id
+		 * @param $post_id
+		 *
+		 * @return mixed
+		 */
+		abstract protected function get_attendees_by_post_id( $post_id );
+
+		/**
+		 * Get Attendees by ticket/attendee ID
+		 *
+		 * @param $attendee_id
+		 *
+		 * @return array
+		 */
+		abstract protected function get_attendees_by_attendee_id( $attendee_id );
+
+		/**
+		 * Get attendees by order id
+		 *
+		 * @param $order_id
+		 *
+		 * @return array
+		 */
+		abstract protected function get_attendees_by_order_id( $order_id );
+
+
+		/**
+		 * Get attendees from provided query
+		 *
+		 * @param WP_Query $attendees_query
+		 * @param          $post_id
 		 *
 		 * @return mixed
 		 */
-		abstract protected function get_attendees( $event_id );
+		abstract protected function get_attendees( WP_Query $attendees_query, $post_id );
```

Comment:
We'll need to make sure that we properly enforce the version dependency of ET+ with ET to make sure this doesn't cause issues.